tictactoe/lib/tictactoe/game.ex

```diff
@@ -1,19 +1,42 @@
 defmodule Tictactoe.Game do
+  alias Tictactoe.Game
+
   defstruct(
     game_state: "playing",
-    current_player: "X",
-    board: [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    current_player: nil,
+    board: [1, 2, 3, 4, 5, 6, 7, 8, 9],
+    player_x: nil,
+    player_o: nil
   )
 
-  def new_game() do
-    %Tictactoe.Game{}
+  def new_game do
+    %Game{} |> setup_players(:human_vs_human)
+  end
+
+  def new_game(mode) do
+    %Game{} |> setup_players(mode)
   end
 
   def make_move(game, position, mark) do
     game
     |> mark_board(position, mark)
     |> evaluate_move
     |> switch_players
+    |> continue
+  end
+
+  def continue(game = %Game{ current_player: player }) do
+    case player["type"] do
+      "human" ->
```

Comment:
Any reason to break those in two line?

It feels like the style could be more consistent with this one
```
%{"mark" => "X"} -> game.player_o
%{"mark" => "O"} -> game.player_x
```